### ISSUE TITLE
Just running an included example (or at least trying to run)

### ISSUE THREAD

**Issue by Antonio-Sorrentini** [eclipse/deeplearning4j#7996]
<pre style="font-size: 9px;">
Any idea why this happens?
It is the included NeuralStyleTransfer example, with no modification at all.
Thank you very much for the help.

org.deeplearning4j.examples.styletransfer.NeuralStyleTransfer
o.n.l.f.Nd4jBackend - Loaded [CpuBackend] backend
o.n.n.NativeOpsHolder - Number of threads used for NativeOps: 6
o.n.n.Nd4jBlas - Number of threads used for BLAS: 6
o.n.l.a.o.e.DefaultOpExecutioner - Backend used: [CPU]; OS: [Windows 10]
o.n.l.a.o.e.DefaultOpExecutioner - Cores: [12]; Memory: [5,3GB];
o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [MKL]
o.d.z.ZooModel - Using cached model at C:\Users\antonio\.deeplearning4j\models\vgg16\vgg16_dl4j_inference.zip
o.d.z.ZooModel - Verifying download...
o.d.z.ZooModel - Checksum local is 3501732770, expecting 3501732770
o.d.n.g.ComputationGraph - Starting ComputationGraph with WorkspaceModes set to [training: NONE; inference: SINGLE], cacheMode set to [NONE]
o.d.e.s.NeuralStyleTransfer - 
========================================================================================================
VertexName (VertexType)           nIn,nOut     TotalParams   ParamsShape                  Vertex Inputs 
========================================================================================================
input_1 (InputVertex)             -,-          -             -                            -             
block1_conv1 (ConvolutionLayer)   3,64         1792          W:{64,3,3,3}, b:{1,64}       [input_1]     
block1_conv2 (ConvolutionLayer)   64,64        36928         W:{64,64,3,3}, b:{1,64}      [block1_conv1]
block1_pool (SubsamplingLayer)    -,-          0             -                            [block1_conv2]
block2_conv1 (ConvolutionLayer)   64,128       73856         W:{128,64,3,3}, b:{1,128}    [block1_pool] 
block2_conv2 (ConvolutionLayer)   128,128      147584        W:{128,128,3,3}, b:{1,128}   [block2_conv1]
block2_pool (SubsamplingLayer)    -,-          0             -                            [block2_conv2]
block3_conv1 (ConvolutionLayer)   128,256      295168        W:{256,128,3,3}, b:{1,256}   [block2_pool] 
block3_conv2 (ConvolutionLayer)   256,256      590080        W:{256,256,3,3}, b:{1,256}   [block3_conv1]
block3_conv3 (ConvolutionLayer)   256,256      590080        W:{256,256,3,3}, b:{1,256}   [block3_conv2]
block3_pool (SubsamplingLayer)    -,-          0             -                            [block3_conv3]
block4_conv1 (ConvolutionLayer)   256,512      1180160       W:{512,256,3,3}, b:{1,512}   [block3_pool] 
block4_conv2 (ConvolutionLayer)   512,512      2359808       W:{512,512,3,3}, b:{1,512}   [block4_conv1]
block4_conv3 (ConvolutionLayer)   512,512      2359808       W:{512,512,3,3}, b:{1,512}   [block4_conv2]
block4_pool (SubsamplingLayer)    -,-          0             -                            [block4_conv3]
block5_conv1 (ConvolutionLayer)   512,512      2359808       W:{512,512,3,3}, b:{1,512}   [block4_pool] 
block5_conv2 (ConvolutionLayer)   512,512      2359808       W:{512,512,3,3}, b:{1,512}   [block5_conv1]
block5_conv3 (ConvolutionLayer)   512,512      2359808       W:{512,512,3,3}, b:{1,512}   [block5_conv2]
block5_pool (SubsamplingLayer)    -,-          0             -                            [block5_conv3]
flatten (PreprocessorVertex)      -,-          -             -                            [block5_pool] 
fc1 (DenseLayer)                  25088,4096   102764544     W:{25088,4096}, b:{1,4096}   [flatten]     
fc2 (DenseLayer)                  4096,4096    16781312      W:{4096,4096}, b:{1,4096}    [fc1]         
predictions (DenseLayer)          4096,1000    4097000       W:{4096,1000}, b:{1,1000}    [fc2]         
--------------------------------------------------------------------------------------------------------
            Total Parameters:  138357544
        Trainable Parameters:  138357544
           Frozen Parameters:  0
========================================================================================================

Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.CommandLineWrapper.main(CommandLineWrapper.java:67)
Caused by: java.lang.IllegalArgumentException: Op.X and Op.Y must have the same data type, but got DOUBLE vs FLOAT
	at org.nd4j.base.Preconditions.throwEx(Preconditions.java:636)
	at org.nd4j.base.Preconditions.checkArgument(Preconditions.java:64)
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(NativeOpExecutioner.java:765)
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(NativeOpExecutioner.java:148)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.addi(BaseNDArray.java:3818)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.addi(BaseNDArray.java:3788)
	at org.deeplearning4j.examples.styletransfer.NeuralStyleTransfer.createCombinationImage(NeuralStyleTransfer.java:192)
	at org.deeplearning4j.examples.styletransfer.NeuralStyleTransfer.transferStyle(NeuralStyleTransfer.java:132)
	at org.deeplearning4j.examples.styletransfer.NeuralStyleTransfer.main(NeuralStyleTransfer.java:124)
	... 5 more

Process finished with exit code 1

**Comment by Antonio-Sorrentini** [comment]
So I reaplced that in the comment with this one and I got passed the problem.
But........ now I get 

Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.CommandLineWrapper.main(CommandLineWrapper.java:67)
Caused by: java.lang.IllegalStateException: Cannot perform backprop: Dropout mask array is absent (already cleared?)

Definitely I would say that this example 
org.deeplearning4j.examples.styletransfer.NeuralStyleTransfer

is NOT suggested if you are new to deeplearning4j. :(

    private INDArray createCombineImageWithRandomPixels() {
        int totalEntries = CHANNELS * HEIGHT * WIDTH;
        float[] result = new float[totalEntries];
        for (int i = 0; i < result.length; i++) {
            result[i] =(float)ThreadLocalRandom.current().nextDouble(-20, 20);
        }
        return Nd4j.create(result, new int[]{1, CHANNELS, HEIGHT, WIDTH});
    }


/*
    private INDArray createCombineImageWithRandomPixels() {
        int totalEntries = CHANNELS * HEIGHT * WIDTH;
        double[] result = new double[totalEntries];
        for (int i = 0; i < result.length; i++) {
            result[i] = ThreadLocalRandom.current().nextDouble(-20, 20);
        }
        return Nd4j.create(result, new int[]{1, CHANNELS, HEIGHT, WIDTH});
    }
*/

**Comment by Antonio-Sorrentini** [comment]
I adjusted it and now it works like a charm! 😀😋😀
I had to modify this method:


<pre>
 private INDArray backPropagate(ComputationGraph vgg16FineTune, INDArray dLdANext, int startFrom)
 {
  for (int i=startFrom;i>0;i--)
  {
   Layer layer=vgg16FineTune.getLayer(ALL_LAYERS[i]);
   dLdANext=layer.backpropGradient(dLdANext, LayerWorkspaceMgr.noWorkspaces()).getSecond();
  }
  return dLdANext;
 }
</pre>
with this other version (added the two lines in the middle):
<pre>
 private INDArray backPropagate(ComputationGraph vgg16FineTune, INDArray dLdANext, int startFrom)
 {
  for (int i=startFrom;i>0;i--)
  {
   Layer layer=vgg16FineTune.getLayer(ALL_LAYERS[i]);

   org.deeplearning4j.nn.conf.layers.Layer conf=(org.deeplearning4j.nn.conf.layers.Layer)layer.getConfig();
   conf.setIDropout(null);

   dLdANext=layer.backpropGradient(dLdANext, LayerWorkspaceMgr.noWorkspaces()).getSecond();
  }
  return dLdANext;
 }
</pre>
You can find the whole modified file with this modification and the previous from the previous post in the attached zip file, 

[NeuralStyleTransfer.zip](https://github.com/eclipse/deeplearning4j/files/3376145/NeuralStyleTransfer.zip)


(it's with a different code format style, you have to reformat it). Is it possible, please, to merge it with the original sources? These problems are extremely frustrating for newbies like me, if solved it's a lot better, they could make the difference between choosing a library or maybe a totally different one and maybe even different language :) 
Thank you anyway.

**Comment by raver119** [comment]
cc @RobAltena @AlexDBlack ^^^

**Comment by agibsonccc** [comment]
Closing — this issue is over 5 years old with no recent activity. The codebase has been significantly rewritten. If this is still relevant, please open a new issue.
